hydra-node/test/Hydra/Chain/Direct/Contract/Mutation.hs

```diff
@@ -728,3 +746,19 @@ replaceHeadId headId = \case
       , Head.headId = headId
       }
   otherState -> otherState
+
+-- | Generates value such that:
+-- - alters between policy id we use in test fixtures with a random one.
+-- - mixing arbitrary token names with 'hydraHeadV1'
+-- - using range from -3 to 3 (excluding 0) for quantity to mimic minting/burning
+genMintedOrBurnedValue :: Gen Value
+genMintedOrBurnedValue = do
+  policyId <-
+    oneof
+      [ headPolicyId <$> arbitrary
+      , pure Fixtures.testPolicyId
+      ]
+  tokenName <- oneof [arbitrary, pure (AssetName $ fromBuiltin hydraHeadV1)]
+  -- simulate minting or burning. 3 tokens should suffice
+  quantity <- oneof (pure <$> [-3, -2, -1, 1, 2, 3])
```

Comment:
Three is quite arbitrary and cannot be controlled from the outside.

Could: use a sized generator like `arbitrarySizedIntegral` (which is the same as `arbitrary :: Int` for example) and excluding `0` using `suchThat` (it's okay if there are clearly more cases that are generated)

That way, quickcheck does test with -1 and 1 _very likely_ and random values up to size. Size is parameter which starts small and grows bigger when more quickcheck tests are run.

Reply:
I wanted to avoid calling `suchThat` but I guess in this case it is fine since we are excluding just one value so it should not choke too much.